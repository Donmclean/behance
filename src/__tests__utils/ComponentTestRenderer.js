import React from 'react';
import { store, history } from '../js/utils/createStore';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import { ConnectedRouter } from 'react-router-redux';

const wrapComponent = (componentToRender) => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Route component={componentToRender} />
        </ConnectedRouter>
    </Provider>
);

export const getFullComponentJSON = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return renderer.create((componentWrapper).toJSON());
};

export const getFullComponentRenderWrapper = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return mount(componentWrapper);
};

export const getShallowComponentRendererWrapper = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return shallow(componentWrapper);
};