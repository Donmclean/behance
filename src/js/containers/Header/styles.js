import React from 'react';
import styled from 'styled-components';
import { Fonts, Colors } from '../../constants/styleConstants';

export const HeaderWrapper = styled.div`
  background: ${Colors.background1};
  height: 100%;
  width: 100%;
  box-shadow: 0px 4px 11px 0px rgba(0,0,0,0.75);
`;

export const HeaderText = styled.h1`
  font-size: ${Fonts.large};
  display: inline-block;
  margin: 30px 0;
  text-align: center;
  width: 100%;
`;
