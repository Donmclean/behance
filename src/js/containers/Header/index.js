import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { HeaderWrapper, HeaderText } from './styles';

class Header extends Component {
    render() {
        return (
            <HeaderWrapper>
                <HeaderText>Coding Challenge</HeaderText>
            </HeaderWrapper> 
        );
    }
}

Header.propTypes = {
    state: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired
};

Header.contextTypes = {
    actions: PropTypes.object
};

const mapStateToProps = (state) => ({state});

export default withRouter(connect(mapStateToProps)(Header));