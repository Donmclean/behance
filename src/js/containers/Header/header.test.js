import React from 'react';
import Header from './index';

import { getFullComponentRenderWrapper } from '../../../__tests__utils/ComponentTestRenderer';

describe('Header Component', () => {
    it('renders <Header /> component successfully in to the DOM with required props', () => {
        const props = {};

        const HeaderWithProps = () => (<Header {...props} />);

        const tree = getFullComponentRenderWrapper(HeaderWithProps);

        expect(tree.debug()).toMatchSnapshot();
    });
});