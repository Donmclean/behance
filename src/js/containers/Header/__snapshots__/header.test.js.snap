// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders <Header /> component successfully in to the DOM with required props 1`] = `
"<Provider store={{...}}>
  <ConnectedRouter history={{...}}>
    <Router history={{...}}>
      <Route component={[Function: HeaderWithProps]}>
        <HeaderWithProps match={{...}} location={{...}} history={{...}} staticContext={[undefined]}>
          <withRouter(Connect(Header))>
            <Route>
              <Connect(Header) match={{...}} location={{...}} history={{...}} staticContext={[undefined]}>
                <Header match={{...}} location={{...}} history={{...}} staticContext={[undefined]} state={{...}} dispatch={[Function]}>
                  <styled.div>
                    <div className=\\"sc-bdVaJa cZNPeo\\">
                      <styled.h1>
                        <h1 className=\\"sc-bwzfXH hcHOGD\\">
                          (Don Mclean)
                        </h1>
                      </styled.h1>
                    </div>
                  </styled.div>
                </Header>
              </Connect(Header)>
            </Route>
          </withRouter(Connect(Header))>
        </HeaderWithProps>
      </Route>
    </Router>
  </ConnectedRouter>
</Provider>"
`;
